ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,169 @@
+import { useSelector } from 'react-redux'
+import { getKnownMethodData } from '../selectors/selectors'
+import { getTransactionActionKey, getStatusKey } from '../helpers/utils/transactions.util'
+import { camelCaseToCapitalize } from '../helpers/utils/common.util'
+import { useI18nContext } from './useI18nContext'
+import { PRIMARY, SECONDARY } from '../helpers/constants/common'
+import { getTokenToAddress } from '../helpers/utils/token-util'
+import { useUserPreferencedCurrency } from './useUserPreferencedCurrency'
+import { formatDateWithYearContext, shortenAddress } from '../helpers/utils/util'
+import {
+  APPROVED_STATUS,
+  SUBMITTED_STATUS,
+  CONTRACT_INTERACTION_KEY,
+  DEPLOY_CONTRACT_ACTION_KEY,
+  INCOMING_TRANSACTION_KEY,
+  TOKEN_METHOD_TRANSFER,
+  TOKEN_METHOD_TRANSFER_FROM,
+  SEND_ETHER_ACTION_KEY,
+  TRANSACTION_CATEGORY_APPROVAL,
+  TRANSACTION_CATEGORY_INTERACTION,
+  TRANSACTION_CATEGORY_RECEIVE,
+  TRANSACTION_CATEGORY_SEND,
+  TRANSACTION_CATEGORY_SIGNATURE_REQUEST,
+  PENDING_STATUS,
+  TOKEN_METHOD_APPROVE,
+} from '../helpers/constants/transactions'
+import { useCurrencyDisplay } from './useCurrencyDisplay'
+import { useTokenDisplayValue } from './useTokenDisplayValue'
+import { useTokenData } from './useTokenData'
+import { tokenSelector } from '../selectors'
+
+// This is duplicated from transactions selectors, additionally
+// from the history view we will already know the pending status
+// of a transaction, but in order to make this hook less coupled
+// to where it is used we should determine the pending status here
+const pendingStatusMap = {
+  [APPROVED_STATUS]: true,
+  [SUBMITTED_STATUS]: true,
+}
+
+const tokenCategoryMap = {
+  [TOKEN_METHOD_APPROVE]: true,
+  [TOKEN_METHOD_TRANSFER]: true,
+  [TOKEN_METHOD_TRANSFER_FROM]: true,
+}
+
+/**
+ * @typedef {Object} TransactionDisplayData
+ * @property {string} title               - primary description of the transaction
+ * @property {string} subtitle            - supporting text describing the transaction
+ * @property {string} category            - the transaction category
+ * @property {string} primaryCurrency     - the currency string to display in the primary position
+ * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
+ * @property {string} status              - the status of the transaction
+ * @property {string} senderAddress       - the Ethereum address of the sender
+ * @property {string} recipientAddress    - the Ethereum address of the recipient
+ */
+
+/**
+ * useTransactionDisplayData
```

Comment:
Nit: can we add a single line description of the function here instead? Maybe extracted from the larger description below.

Reply:
```
/**
 * Get computed values used for displaying transaction data to a user
 *
```